types/react/test/index.ts

```diff
@@ -519,13 +519,17 @@ declare const mixedChildren: Array<JSX.Element | string>;
 declare const singlePluralChildren: JSX.Element | JSX.Element[];
 declare const renderPropsChildren: () => JSX.Element;
 
+// $ExpectType null
+const mappedChildrenArray0 = React.Children.map(null, num => num);
+// $ExpectType undefined
+const mappedChildrenArray1 = React.Children.map(undefined, num => num);
```

Comment:
> If children is null or undefined, this method will return null or undefined rather than an array.

https://reactjs.org/docs/react-api.html#reactchildrenmap